src/enforcer/HolographRoyalties.sol

```diff
@@ -597,4 +650,31 @@ contract HolographRoyalties is Admin, Owner, Initializable {
   function getTokenAddress(string memory tokenName) public view returns (address) {
     return _getTokenAddress(tokenName);
   }
+
+  /**
+   * @notice Used to wrap function calls to check if they return without revert regardless of return type.
+   * @dev
+   * @return Returns true if the wrapped function call returns without a revert even if it doesn't return true.
+   */
+  function _callOptionalReturn(
```

Comment:
In order to accommodate non-compliant ERC-20 tokens (like USDT), which do not return a bool on success, we wrap the ERC20 calls. This allows to not revert when a call returns no data. The logic works by: making a call, checking if return opcode is a revert, if it is then it reverts as well, if it's not then it checks for return data, if return data exists, it is returned as a bool, if return data does not exist (0 length) then success is expected and returns true.